packages/react-core/src/components/Page/PageSection.tsx

```diff
@@ -41,21 +53,46 @@ const variantStyle = {
   [PageSectionVariants.darker]: styles.modifiers.dark_100
 };
 
+const paddingBreakpoints = {
+  [PageSectionBreakpoints.sm]: styles.modifiers.paddingOnSm,
+  [PageSectionBreakpoints.md]: styles.modifiers.paddingOnMd,
+  [PageSectionBreakpoints.lg]: styles.modifiers.paddingOnLg,
+  [PageSectionBreakpoints.xl]: styles.modifiers.paddingOnXl,
+  [PageSectionBreakpoints['2xl']]: styles.modifiers.paddingOn_2xl
+};
+
+const noPaddingBreakpoints = {
+  [PageSectionBreakpoints.sm]: styles.modifiers.noPaddingOnSm,
+  [PageSectionBreakpoints.md]: styles.modifiers.noPaddingOnMd,
+  [PageSectionBreakpoints.lg]: styles.modifiers.noPaddingOnLg,
+  [PageSectionBreakpoints.xl]: styles.modifiers.noPaddingOnXl,
+  [PageSectionBreakpoints['2xl']]: styles.modifiers.noPaddingOn_2xl
+};
+
 export const PageSection: React.FunctionComponent<PageSectionProps> = ({
   className = '',
   children,
   variant = 'default',
   type = 'default',
-  noPadding = false,
+  hasNoPadding = false,
+  hasPaddingOn,
+  hasNoPaddingOn,
   isFilled,
   ...props
 }: PageSectionProps) => (
-  // TODO: Implement https://github.com/patternfly/patternfly/pull/2816
   <section
     {...props}
     className={css(
       variantType[type],
-      noPadding && styles.modifiers.noPadding,
+      hasNoPadding && styles.modifiers.noPadding,
+      hasPaddingOn &&
+        (hasPaddingOn.constructor !== Array
```

Comment:
We have helpers for this! Checkout https://github.com/patternfly/patternfly-react/blob/v4/packages/react-core/src/helpers/util.ts#L243 and follow what Flex or Toolbar do for example.

Reply:
I had seen them but didn't use them as PageSection only needed breakpoints on padding and no other modifiers, but its probably better this way for any future expansion on breakpoint specific modifiers. I'll update.